apps/entry/filter_set.py

```diff
@@ -669,6 +649,29 @@ def authoring_organization_types_filter(self, qs, name, value):
             return qs.filter(organization_types__in=value).distinct()
         return qs
 
+    def filter_commented_entries(self, queryset, name, value):
+        if value:
+            return queryset.filter(
+                review_comments__comment_type__in=[
+                    EntryReviewComment.CommentType.COMMENT,
+                    EntryReviewComment.CommentType.UNVERIFY,
+                    EntryReviewComment.CommentType.UNCONTROL,
+                ]
+            )
+        return queryset.filter(
+            models.Q(
+                review_comments__comment_type__in=[
+                    EntryReviewComment.CommentType.VERIFY,
+                    EntryReviewComment.CommentType.CONTROL,
+                ]
+            ) | models.Q(review_comments__isnull=True)
+        )
```

Comment:
- Entry 1:
    - Stage 1: Nothing
        - comment: True: Hide
        - comment: False: Show
    - Stage 2: Add a comment
        - comment: True: Show
        - comment: False: Hide
    - Stage 1: Add a verified comment
        - comment: True: Show
        - comment: False: Show <-> Hide
    - Stage 1: Add a comment
        - comment: True: Show
        - comment: False: Show <-> Hide
        
- Entry 2:
    - Stage 1: Nothing
        - comment: True: Hide
        - comment: False: Show
    - Stage 1: Add a verified comment
        - comment: True: Hide
        - comment: False: Show
    - Stage 1: Add a comment
        - comment: True: Show
        - comment: False: Show <-> Hide

```suggestion
   _filter = dict(
            review_comments__comment_type__in=[
                    EntryReviewComment.CommentType.COMMENT,
                    EntryReviewComment.CommentType.UNVERIFY,
                    EntryReviewComment.CommentType.UNCONTROL,
            ]
        )
        if value:
            return queryset.filter(**_filter)
        return queryset.exclude(**_filter)
```